Notebooks/PBF Analysis.Rmd

```diff
@@ -0,0 +1,216 @@
+---
+output: html_document
+--- 
+## **Philaldelphia Bail Fund** 
+
+\right<span style="color: grey;">July 23, 2020</span>\right  
+
+  [GitHub](https://github.com/CodeForPhilly/pbf-analysis)  
+  [Website](https://www.phillybailfund.org/)
+  
+
+***
+
+#### _**Table of Contents**_
+* [Introduction](#**Introduction**) 
+* [Data Cleaning](#**Data Cleaning**) 
+* [Text Preprocessing](#**Text Preprocessing**) 
+* [Predictive Modeling](#**Predictive Modeling**) 
+* [Analysis](#**Analysis**) 
+* [Ideas for Further Analysis](#**Ideas for Further Analysis**)
+
+***
+
+## <a id="**Introduction**"></a>**Introduction**
+[]
+
+## <a id="**Data Cleaning**"></a>**Data Cleaning**
+```{r message = FALSE, warning = FALSE}
+library(dplyr)
+library(corrplot)
+library(Metrics)
+library(gbm)
+library(ggplot2)
+library(stringr)
+library(tm) 
+library(textstem)
+library(tidytext)
+library(wordcloud)
+```
+##### A1. Import data
+```{r}
+url <- "https://raw.githubusercontent.com/CodeForPhilly/pbf-analysis/master/Data/0c_distinct_dockets.csv"
+data <- read.csv(url)
+head(data, 10)
+```
+
+##### A2. Delete rows where bail_type = 'Denied' and bail_status = "ROR" or "Nonmonetary" 
+```{r}
+data <- data %>% mutate(bail_status = as.character(bail_status),
+                        bail_type = as.character(bail_type)) %>% 
+  filter(!bail_status %in% c("Nonmonetary", "ROR"), 
+         !bail_type %in% c("Denied"))
+```
+
+##### A3. Create hour of day and day of week from filing_date
+```{r}
+data$date <- as.Date(data$filing_date)
+data$day_of_week <- as.factor(weekdays(data$date))
+data$time <- as.numeric(substr(data$filing_date, 12, 13))
+data$bail_status <- as.factor(data$bail_status)
+```
+
+###### A4. Impute age with average of age
+```{r}
+avg.age <- mean(data[-which(is.na(data)),]$age)
+data <- data %>% mutate(age = ifelse(is.na(age) == TRUE, avg.age, age))
+
+ggplot(data, aes(x=date, y=bail_amount)) +
+  geom_line( color="steelblue") + 
+  xlab("")
+```
+
+##### A5. Keep = bail_amount, charge, day of week, hour of day, age, represented_by, address
+
+  
+
+  
+
+## <a id="**Text Preprocessing**"></a>**Text Preprocessing**
+##### B1. Clean text
+```{r}
+data$charge = tolower(data$charge)
+
+new_stopwords <- c(stopwords('en'), "criminal", "crim", "cause", "intent", "int", "attempt")
+data$charge = gsub(paste(new_stopwords, collapse = '\\b|\\b'), '', data$charge)
+data$charge = gsub('[[:digit:]]', '', data$charge)
+data$charge = gsub('[[:punct:]]', '', data$charge)
+data$charge = gsub('\\s+',' ', data$charge)
+```
+
+##### B2. Group charges - reduce from 136 to 97 levels
+```{r}
+#data$charge <- as.character(data$charge)
+data$charge <- ifelse(str_detect(data$charge, "burglary"), "burglary",

```

Comment:
`case_then` might be a good way to flatten these ifelse calls :)

https://dplyr.tidyverse.org/reference/case_when.html